src/util/jwt.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Checks if a given access token is a JWT.
+ *
+ * @param {string} accessToken - The access token to check
+ * @returns {boolean} True if the token is a valid JWT token, false otherwise.
+ */
+export function isJWTToken(accessToken: string): boolean {
+  const parts = accessToken.split('.');
+  if (parts.length !== 3) return false;
+
+  const header = parts[0];
+  try {
+    JSON.parse(atob(header));
```

Comment:
JWTs have 3 base64-encoded parts (header,payload and signature) separated by a dot.

if we got here, it could also be an opaque token that has 3 dots in it so we do a final check by decoding the base64 header -> try to `JSON.parse` (if the parse didn't fail then we assume it's a JWT header in JSON and return true)